File: AtCoder/AGC026/A.cpp
#include <iostream>

using namespace std;

int main(){
    int N; cin >> N;
    int res = 0;
    int cnt = 0;
    int prev = -1;
    for(int i=0;i<N;i++){
        int a; cin >> a;
        if(a != prev){
            res += cnt/2;
            cnt = 1;
        } else {
            ++cnt;
        }
        prev = a;
    }
    res += cnt/2;
    cout << res << endl;
}

File: AtCoder/AGC026/src/bin/a.rs
use proconio::input;

fn main() {
    input! {
        n: usize,
        a: [usize; n]
    }
    let mut prev = 0;
    let mut cnt = 0;
    let mut res = 0;
    for &c in &a {
        if c == prev {
            cnt += 1;
            res += cnt % 2;
        } else {
            cnt = 0;
            prev = c;
        }
    }
    println!("{}", res);
}
